### PR TITLE
Edited streetcode and added handlers

### PR DESCRIPTION
**Pull request by EyR1oN**
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)

## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions

### CODE REVIEW

**Review by PingvinAustr — APPROVED** [review]

File: Streetcode/Streetcode.BLL/Mapping/AdditionalContent/TagProfile.cs [hunk 1]
```diff
@@ -8,6 +8,6 @@ public class TagProfile : Profile
 {
     public TagProfile()
     {
-        CreateMap<Tag, TagDTO>().ReverseMap();
+        CreateMap<Tag, TagDTO>().ForMember(x => x.Streetcodes, conf => conf.Ignore());
```
Comment (AleXLaeR) [inline Streetcode/Streetcode.BLL/Mapping/AdditionalContent/TagProfile.cs:11]: Why we should ignore streetcodes p-ty mapping?

File: Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Handlers/QueryHandlers/GetStreetcodeById.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        _repositoryWrapper = repositoryWrapper;
+        _mapper = mapper;
+    }
+
+    public async Task<Result<StreetcodeDTO>> Handle(GetStreetcodeByIdQuery request, CancellationToken cancellationToken)
+    {
+        var streetcode = await _repositoryWrapper.StreetcodeRepository.GetSingleOrDefaultAsync(
+            predicate: st => st.Id == request.id,
+            include: source => source
+                    .Include(l => l.Tags));
```
Comment (AleXLaeR) [inline Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Handlers/QueryHandlers/GetStreetcodeById.cs:31]: Please handle the following unhappy paths:
- **SingleOrDefault** may throw an _**InvalidOperationException**_ when there is more than one element that matches predicate; On the other hand, if that condition will never be reached, why we need to iterate over an entire Set?
- If **default value** is returned (__null__), IMapper will throw an exception which will find its way up to the UI. Do we want that behaviour?

File: Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Queries/GetStreetcodeByIdQuery.cs [hunk 3]
```diff
@@ -0,0 +1,10 @@
+﻿using FluentResults;
+using MediatR;
+using Streetcode.BLL.DTO.Streetcode;
+using Streetcode.BLL.DTO.Streetcode.TextContent;
+using Streetcode.BLL.DTO.Streetcode.Types;
+
+namespace Streetcode.BLL.MediatR.Streetcode.Streetcode.Queries
```
Comment (AleXLaeR) [inline Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Queries/GetStreetcodeByIdQuery.cs:7]: Please switch to file-scoped namespace.

File: Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Handlers/QueryHandlers/GetStreetcodeById.cs [hunk 4]
```diff
@@ -0,0 +1,36 @@
+﻿using AutoMapper;
```
Comment (AleXLaeR) [inline Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Handlers/QueryHandlers/GetStreetcodeById.cs:1]: Also please consider to switch onto the file structure which was provided in my example (Folder per action, Query/Command and Handler go inside of it).

**Review by AleXLaeR — CHANGES_REQUESTED** [review]
Pelase review my comments.

File: Streetcode/Streetcode.BLL/Mapping/AdditionalContent/TagProfile.cs (hunk 1 above)
Comment (EyR1oN) [reply Streetcode/Streetcode.BLL/Mapping/AdditionalContent/TagProfile.cs:11]: Because we have infinite loop in our dto.

Comment (dimaln) [inline Streetcode/Streetcode.BLL/Mapping/AdditionalContent/TagProfile.cs:11]: what about mapping TagDto -> Tag?
if we use it in any place , please add `CreateMap<TagDTO, Tag>();`

File: Streetcode/Streetcode.BLL/Mapping/Streetcode/Types/EventStreetCodeProfile.cs [hunk 5]
```diff
@@ -8,6 +10,6 @@ public class EventStreetcodeProfile : Profile
 {
     public EventStreetcodeProfile()
     {
-        CreateMap<EventStreetcode, EventStreetcodeDTO>().ReverseMap();
+        CreateMap<EventStreetcode, EventStreetcodeDTO>().IncludeBase<StreetcodeContent, StreetcodeDTO>().ReverseMap();
```
Comment (dimaln) [inline Streetcode/Streetcode.BLL/Mapping/Streetcode/Types/EventStreetCodeProfile.cs:13]: please separate lines

File: Streetcode/Streetcode.BLL/Mapping/Streetcode/Types/PersonStreetCodeProfile.cs [hunk 6]
```diff
@@ -8,6 +10,6 @@ public class PersonStreetcodeProfile : Profile
 {
     public PersonStreetcodeProfile()
     {
-        CreateMap<PersonStreetcode, PersonStreetcodeDTO>().ReverseMap();
+        CreateMap<PersonStreetcode, PersonStreetcodeDTO>().IncludeBase<StreetcodeContent, StreetcodeDTO>().ReverseMap();
```
Comment (dimaln) [inline Streetcode/Streetcode.BLL/Mapping/Streetcode/Types/PersonStreetCodeProfile.cs:13]: separate lines for readability

File: Streetcode/Streetcode.BLL/Mapping/AdditionalContent/TagProfile.cs (hunk 1 above)
Comment (EyR1oN) [reply Streetcode/Streetcode.BLL/Mapping/AdditionalContent/TagProfile.cs:11]: For now, we use only get request, so mapping into entity is not used. Someone will add it when need it.

Comment (AleXLaeR) [reply Streetcode/Streetcode.BLL/Mapping/AdditionalContent/TagProfile.cs:11]: If that's the case, what you've written is a temporary stub. I am sure there exists a way to map all properties correctly (consider reviewing other profiles addressing this issue).

Comment (AleXLaeR) [reply Streetcode/Streetcode.BLL/Mapping/AdditionalContent/TagProfile.cs:11]: During initial db-setup, I also encoutered infinite cycles in model relationships, however, I managed to find a way to fix the latter.

**Review by dimaln — CHANGES_REQUESTED** [review]

File: Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Handlers/QueryHandlers/GetStreetcodeById.cs (hunk 2 above)
Comment (dimasster) [reply Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Handlers/QueryHandlers/GetStreetcodeById.cs:31]: We have never get more than 1 streetcodes with following id, so why we have to handle this exception?
Added null exception handler

File: Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Queries/GetStreetcodeByIdQuery.cs (hunk 3 above)
Comment (dimasster) [reply Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Queries/GetStreetcodeByIdQuery.cs:7]: done

File: Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Handlers/QueryHandlers/GetStreetcodeById.cs (hunk 4 above)
Comment (dimasster) [reply Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Handlers/QueryHandlers/GetStreetcodeById.cs:1]: done

File: Streetcode/Streetcode.BLL/Mapping/Streetcode/Types/EventStreetCodeProfile.cs (hunk 5 above)
Comment (dimasster) [reply Streetcode/Streetcode.BLL/Mapping/Streetcode/Types/EventStreetCodeProfile.cs:13]: done

File: Streetcode/Streetcode.BLL/Mapping/Streetcode/Types/PersonStreetCodeProfile.cs (hunk 6 above)
Comment (dimasster) [reply Streetcode/Streetcode.BLL/Mapping/Streetcode/Types/PersonStreetCodeProfile.cs:13]: done

File: Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Handlers/QueryHandlers/GetStreetcodeById.cs (hunk 2 above)
Comment (AleXLaeR) [reply Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Handlers/QueryHandlers/GetStreetcodeById.cs]: Please carefully review the first bullet on the list.

**Review by AleXLaeR — APPROVED** [review]

File: Streetcode/Streetcode.BLL/Mapping/AdditionalContent/TagProfile.cs (hunk 1 above)
Comment (dimasster) [reply Streetcode/Streetcode.BLL/Mapping/AdditionalContent/TagProfile.cs:11]: done

File: Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Handlers/QueryHandlers/GetStreetcodeById.cs (hunk 2 above)
Comment (dimasster) [reply Streetcode/Streetcode.BLL/MediatR/Streetcode/Streetcode/Handlers/QueryHandlers/GetStreetcodeById.cs]: done

**Review by Tatiana2424 — APPROVED** [review]

**Pull request merged**